test/code_sponsor/impressions/impressions_test.exs

```diff
@@ -1,99 +1,114 @@
-
-#     test "update_impression/2 with invalid data returns error changeset" do
-#       impression = impression_fixture()
-#       assert {:error, %Ecto.Changeset{}} = Impressions.update_impression(impression, @invalid_attrs)
-#       assert impression == Impressions.get_impression!(impression.id)
-#     end
-
-#     test "delete_impression/1 deletes the impression" do
-#       impression = impression_fixture()
-#       assert {:ok, %Impression{}} = Impressions.delete_impression(impression)
-#       assert_raise Ecto.NoResultsError, fn -> Impressions.get_impression!(impression.id) end
-#     end
-
-#     test "change_impression/1 returns a impression changeset" do
-#       impression = impression_fixture()
-#       assert %Ecto.Changeset{} = Impressions.change_impression(impression)
-#     end
-#   end
-# end
+defmodule CodeSponsor.ImpressionsTest do
+  use CodeSponsor.DataCase
+
+  alias CodeSponsor.Impressions
+
+  describe "impressions" do
+    alias CodeSponsor.Schema.Impression
+
+    @valid_attrs %{browser: "some browser", city: "some city", country: "some country", device_type: "some device_type", ip: "some ip", latitude: "120.5", longitude: "120.5", os: "some os", postal_code: "some postal_code", region: "some region", screen_height: 42, screen_width: 42, user_agent: "some user_agent", utm_campaign: "some utm_campaign", utm_content: "some utm_content", utm_medium: "some utm_medium", utm_source: "some utm_source", utm_term: "some utm_term"}
+    @update_attrs %{browser: "some updated browser", city: "some updated city", country: "some updated country", device_type: "some updated device_type", ip: "some updated ip", latitude: "456.7", longitude: "456.7", os: "some updated os", postal_code: "some updated postal_code", region: "some updated region", screen_height: 43, screen_width: 43, user_agent: "some updated user_agent", utm_campaign: "some updated utm_campaign", utm_content: "some updated utm_content", utm_medium: "some updated utm_medium", utm_source: "some updated utm_source", utm_term: "some updated utm_term"}
+    @invalid_attrs %{browser: nil, city: nil, country: nil, device_type: nil, ip: nil, latitude: nil, longitude: nil, os: nil, postal_code: nil, region: nil, screen_height: nil, screen_width: nil, user_agent: nil, utm_campaign: nil, utm_content: nil, utm_medium: nil, utm_source: nil, utm_term: nil}
+
+    test "list_impressions/0 returns all impressions" do
+      impression = CodeSponsor.Support.Fixture.generate(:impression)
+      assert Impressions.list_impressions() == [impression]
+    end
+
+    test "get_impression!/1 returns the impression with given id" do
+      impression = CodeSponsor.Support.Fixture.generate(:impression)
+      assert Impressions.get_impression!(impression.id) == impression
+    end
+
+    test "create_impression/1 with valid data creates a impression" do
+      property = CodeSponsor.Support.Fixture.generate(:property)
+      valid_attrs = @valid_attrs  |> Map.merge(%{property_id: property.id} )
```

Comment:
Places like this would normally be handled by a similar put_assoc like in Property, as an FYI - @coderberry